src/graph/src/modifiers.ts

```diff
@@ -303,9 +317,56 @@ export class GraphModifier {
         /* c8 ignore next - difficult to test branch */
         this.#activeEntries.get(initial)?.get(name)?.get(from) ??
         this.newModifier(from, initial)
-      res.add(initialEntry)
+      all.set(initialEntry.modifier.breadcrumb, initialEntry)
+    }
+    // selects the active entry that should apply to this dependency,
+    // any active entry that is done parsing has the priority, if we
+    // find multiple entries then we use css specificity to pick a winner
+    // if we have multiple matches but no active entry is complete, then
+    // we pick the one with the highest specificity breadcrumb
+    const arr = [...all.values()]
+    const completeEntries = arr.filter(
+      active =>
+        active.interactiveBreadcrumb.current ===
+        active.modifier.breadcrumb.last,
+    )
+    // deregister completed entries
+    for (const entry of completeEntries) {
+      this.deregisterModifier(entry.modifier)
+    }
+    // returns the highest specificity entry from either the complete entries
+    // if any were found or from any of the entries if available, otherwise
+    // it will return undefined as no entry is found in the `all` map
+    return !completeEntries.length ?
+        all.get(
+          specificitySort(arr.map(i => i.modifier.breadcrumb))[0],
+        )
+      : all.get(
+          specificitySort(
+            completeEntries.map(i => i.modifier.breadcrumb),
+          )[0],
+        )
```

Comment:
```suggestion
    const entries = completeEntries.length ? completeEntries : arr
    return all.get(
      specificitySort(entries.map(i => i.modifier.breadcrumb))[0],
    )
```

Is this equivalent? If so, I find this a little more readable since the branching made me think more was going to be differentiated instead of only switching `arr` vs `completeEntries`

Reply:
good call! I think so!